pkg/manager/manager.go

```diff
@@ -494,6 +494,11 @@ func (o Options) AndFromOrDie(loader config.ControllerManagerConfiguration) Opti
 }
 
 func (o Options) setLeaderElectionConfig(obj v1alpha1.ControllerManagerConfigurationSpec) Options {
+	if obj.LeaderElection == nil {
```

Comment:
There is a nilcheck right below this (`!o.LeaderElection && obj.LeaderElection.LeaderElect != nil`

Can you show a stacktrace of this panicing and please add a test? The current fix doesn't seem to me like it will do anything

Reply:
The nilcheck right below is the line that fails for me, since `obj.LeaderElection` is nil (and therefore `obj.LeaderElection.LeaderElect` panics). I'll see if I can figure out where to add test code that executes this path.

Reply:
@alvaroaleman I pushed a test that panics without this check.

Reply:
@alvaroaleman The `!o.LeaderElection && obj.LeaderElection.LeaderElect != nil` is ok, but the logic below like `if o.LeaderElectionResourceLock == "" && obj.LeaderElection.ResourceLock != "" {` will panic if `obj.LeaderElection` is nil.